File: control-assurance-sp/src/types/GoElement.ts
import { IEntity } from "./Entity";
import { IGoForm } from "./GoForm";
import { IGoDefElement } from "./GoDefElement";
import { IGoAssignment } from "./GoAssignment";

export interface IGoElement extends IEntity {

    GoFormId?: number;
    GoDefElementId?: number;
    EvidenceStatement?: string;
    Rating?: string;
    CompletionStatus?: string;
    MarkReadyForApproval?:boolean;
    GoForm?: IGoForm;
    GoDefElement?: IGoDefElement;

    GoAssignments?: IGoAssignment[];

}

export class GoElement implements IGoElement{ 
    public ID: number = 0;
    public Title: string = null;
    public GoFormId:number = 0;
    public GoDefElementId:number = 0;
    public EvidenceStatement?: string = null;
    public Rating?: string = null;
    public CompletionStatus?: string = null;
    public MarkReadyForApproval?:boolean = null;
    public GoForm?: IGoForm = null;
    public GoDefElement?: IGoDefElement = null;

    public GoAssignments = [];

    constructor(goFormId: number, goDefElementId: number) {
        this.GoFormId = goFormId;
        this.GoDefElementId = goDefElementId;
    }
}

File: ControlAssuranceAPI/Models/GoElement.cs
using System;
using System.Collections.Generic;

namespace CAT.Models;

public partial class GoElement
{
    public int ID { get; set; }

    public string? Title { get; set; }

    public int? GoFormId { get; set; }

    public int? GoDefElementId { get; set; }

    public string? EvidenceStatement { get; set; }

    public string? Rating { get; set; }

    public string? CompletionStatus { get; set; }

    public bool? MarkReadyForApproval { get; set; }

    public string? GoElementChangeLog { get; set; }

    public virtual ICollection<GoAssignment> GoAssignments { get; } = new List<GoAssignment>();

    public virtual GoDefElement? GoDefElement { get; set; }

    public virtual ICollection<GoElementAction> GoElementActions { get; } = new List<GoElementAction>();

    public virtual ICollection<GoElementEvidence> GoElementEvidences { get; } = new List<GoElementEvidence>();

    public virtual ICollection<GoElementFeedback> GoElementFeedbacks { get; } = new List<GoElementFeedback>();

    public virtual GoForm? GoForm { get; set; }
}
